server/jetstream.go

```diff
@@ -427,25 +427,44 @@ func (s *Server) configAllJetStreamAccounts() error {
 	s.mu.Lock()
 	// Bail if server not enabled. If it was enabled and a reload turns it off
 	// that will be handled elsewhere.
-	if s.js == nil {
+	js := s.js
+	if js == nil {
 		s.mu.Unlock()
 		return nil
 	}
 
 	var jsAccounts []*Account
-
 	s.accounts.Range(func(k, v interface{}) bool {
 		jsAccounts = append(jsAccounts, v.(*Account))
 		return true
 	})
+	accounts := &s.accounts
 	s.mu.Unlock()
 
-	// Process any jetstream enabled accounts here.
+	// Process any jetstream enabled accounts here. These will be accounts we are
+	// already aware of at startup etc.
 	for _, acc := range jsAccounts {
 		if err := s.configJetStream(acc); err != nil {
 			return err
 		}
 	}
+
+	// Now walk all the storage we have and resolve any accounts that we did not process already.
+	// This is important in resolver/operator models.
+	fis, _ := ioutil.ReadDir(js.config.StoreDir)
+	for _, fi := range fis {
+		if accName := fi.Name(); accName != _EMPTY_ {
+			// Only load up ones not already loaded since they are processed above.
+			if _, ok := accounts.Load(accName); !ok {
+				if acc, err := s.lookupAccount(accName); err != nil {
```

Comment:
so accounts we cant lookup here will later lazy load and enable? say operator gets it from some other server into its cache etc?

Reply:
Yes we will load on-demand all accounts that are represented by stable storage at startup.